resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/submit/KubernetesClientApplication.scala

```diff
@@ -133,6 +134,17 @@ private[spark] class Client(
       .build()
     val driverPodName = resolvedDriverPod.getMetadata.getName
 
+    val preKubernetesResources = resolvedDriverSpec.preKuberenetesResources
```

Comment:
This is for the PodGroup resource creation, previously, feature step only have the ability to create the post ready resource after pod create. But we still have some case that the resource should be ready before pods.

And looks like it could be a separated PR.

Reply:
https://github.com/apache/spark/pull/34599